### DISCUSSION TITLE
I get same score between euclidean distance and cosine similarity for all questions

### DISCUSSION THREAD

**Discussion by stepkurniawan** [facebookresearch/faiss#3217] [Q&A]
Hi all, 

Im using LangChain x FAISS. And everytime I compare the distance score between the default euclidean and cosine, it will always give me the same score, so I dont know maybe I did something wrong (or not). Could you check please?

This is how I change the distance metrics: 
```
faiss_distance_strategy = get_faiss_distance_strategy(self.index_distance) # can be default eucledian, or DistanceStrategy.MAX_INNER_PRODUCT, or DistanceStrategy.COSINE

# langchain from_documents
self.db = self.vectorstore_obj.from_documents(split_docs, self.embedding, distance_strategy=faiss_distance_strategy) 

self.db.save_local(self.save_path)
```

and this is how I check the distance score: 
```
similarity_search_with_score(query, k=top_k)
```
and here is the problem, everytime I get a result, for example, I get a document A, with distance 0.39182765 using euclidean, I also got the same score with cosine, even down to the 8th digits behind comma. With MAX_INNER_PRODUCT i get something different, probably because its searching for maximum, and not minimum. 

is it because everything got normalized? 

Thank you!

**Comment by algoriddle** [comment]
This would be a better question for langchain: https://github.com/langchain-ai/langchain/issues

**Comment by xrd** [comment]
In the README, it does indicate normalization plays a part for cosine similarity.

https://github.com/facebookresearch/faiss?tab=readme-ov-file#introduction
